src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -112,7 +112,7 @@ public Task StartAsync()
                     {
                         _startTcs.SetResult(null);
                     }
-                });
+                }).ForceAsync();
```

Comment:
Why not following the `XXXCore` pattern? This is fragile and can be easily changed incorrectly during a refactoring. 

Also, I think the code still has an issue you are trying to fix - I think when you call `StartAsync().Wait()` on a stopped client will deadlock because it will try to run on user's code. Using `XXXCore` pattern should fix that.